src/lib.rs

```diff
@@ -29,33 +29,36 @@ use core::{any::Any, fmt::Debug};
 
 /// Result of a validation
 ///
-/// The result is `Ok` and empty if the validation succeeded
-/// or otherwise a validation context with one or more
-/// violations wrapped into `Err`.
+/// The result is `Ok` and empty if the validation succeeded. It is
+/// a validation context wrapped into `Err` that carries one or more
+/// invalidities.
+///
+/// In contrast to common results the actual payload is carried by
+/// the error variant while a successful result is just the unit type.
 pub type Result<V> = core::result::Result<(), Context<V>>;
 
-/// Validation objectives that might be violated
-///
-/// A validation fails if one or more objectives are violated.
+/// Invalidities that cause validation failures
 ///
-/// These types are typically `enum`s with one variant per objective.
-/// Some of the variants may recursively wrap dependent validations
-/// to trace back the root cause.
-pub trait Validation: Any + Debug {}
+/// Validations fail if one or more objectives are considered invalid.
+/// These invalidity objectives are typically represented by sum types
+/// (`enum`) with one variant per objective. Some of the variants may
+/// recursively wrap an invalidity of a subordinate validation to trace
+/// back root causes.
+pub trait Invalidity: Any + Debug {}
```

Comment:
The description above is easy to understand but it's a bit difficult to comprehend the connection to `Any` and `Debug`. And I think most users aren't used to associated trait types.
So maybe one ore two more sentences would be helpful to understand what added value can be expected with this approach.